### DISCUSSION TITLE
Storage and display on the website.

### DISCUSSION THREAD

**Discussion by ZigglerPL** [community/community#64339] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hello, where should I start to be able to achieve the effect where the entire element's dom structure is stored in global memory and displayed on the page? I mainly learn javascript, css and html.

**Comment by zainmujahid** [comment]
Hello! It's fantastic that you're keen on exploring this concept. To achieve the effect of storing an element's entire DOM structure and displaying it on a page, follow these steps:

1. **HTML:** Start with a basic HTML structure and create a container element for display.

2. **JavaScript:** Construct the DOM structure using `document.createElement()` and store it globally.

3. **Rendering:** Use JavaScript to render the stored structure into the container element.

4. **Optional Styling:** Apply CSS for aesthetics using inline styles or a separate stylesheet.

5. **Considerations:** While this approach is instructive, consider modern frameworks like React for more complex applications.

Enjoy your exploration into web development!

**Reply by ZigglerPL** [reply]
Thanks!

**Comment by mutasim77** [comment]
It sounds like you're interested in manipulating the DOM (Document Object Model) using JavaScript to store and display elements globally on a web page. This can be achieved by following these steps:
- 1 HTML Structure: Create the HTML structure you want to work with. This could be a container element where you want to display your dynamic content.

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Global DOM Example</title>
</head>
<body>
    <div id="content-container">
        <!-- Your content will be displayed here -->
    </div>
    <button id="add-element">Add Element</button>
    <script src="script.js"></script>
</body>
</html>
```

- 2 JavaScript: Create a JavaScript file (e.g., script.js) and link it in your HTML file. In this script, you'll manipulate the DOM to add and display elements globally.

```js
// script.js

// Get the container element
const contentContainer = document.getElementById('content-container');

// Get the button element
const addButton = document.getElementById('add-element');

// Function to add and display elements
function addElement() {
    // Create a new element
    const newElement = document.createElement('div');
    newElement.className = 'dynamic-element';
    newElement.textContent = 'New Dynamic Element';

    // Append the new element to the container
    contentContainer.appendChild(newElement);
}

// Add event listener to the button
addButton.addEventListener('click', addElement);
```

Now, when you click the "Add Element" button, the JavaScript function will create a new <div> element and append it to the container in the DOM. The newly added elements will be stored in memory and displayed on the page.

This basic example demonstrates the concept of manipulating the DOM using JavaScript to dynamically add elements. You can extend this by adding more complex functionality, such as removing elements, updating content, or applying more sophisticated styling.

**Reply by ZigglerPL** [reply]
I know this method and use it often, but I meant more on the basis of actions on the server.
